opacclient/opacapp/src/main/java/de/geeksfactory/opacclient/reminder/SyncAccountAlarmListener.java

```diff
@@ -0,0 +1,33 @@
+package de.geeksfactory.opacclient.reminder;
+
+import com.commonsware.cwac.wakeful.WakefulIntentService;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.Context;
+
+public class SyncAccountAlarmListener implements WakefulIntentService.AlarmListener {
+    // We could use a longer interval like 3 hours, but on API < 19, only the AlarmManager
+    // .INTERVAL_... constants make the AlarmManager use inexact alarm delivery for better
+    // battery efficiency. The next larger constant available would be AlarmManager
+    // .INTERVAL_HALF_DAY.
```

Comment:
We really should discuss this, as it greatly affects both battery life and mobile data tariffs as well as the quality of the notification service.

During normal operation, half a day *should* really be enough, shouldn't it? I've never heard of a library with lending times less than a week, so if you lend out a new item, it should be enough for all notification purposes if the app only sees it half a day later.

A half day is **not enough** if a sync attempts fail, either because of a network connection failure or because the notification service has been limited to WiFi and WiFi is not available, because "half day" is long enough to coincidentally always occur when I'm not in a WiFi network. However, in this case we could go without inexact repetition.

So if I do not miss anything, we should absolutely get away with *regular* runs every (inexact) half day and *repeated runs due to failures* at fixed intervals (e.g. 1 hour after the failure).